apollo-runtime/src/main/java/com/apollographql/apollo/cache/normalized/NormalizedCache.java

```diff
@@ -102,6 +102,10 @@
     checkNotNull(recordSet, "recordSet == null");
     checkNotNull(cacheHeaders, "cacheHeaders == null");
 
+    if (cacheHeaders.hasHeader(ApolloCacheHeaders.DO_NOT_STORE)) {
+      return Collections.emptySet();
+    }
+
```

Comment:
I believe we need to move this condition up to and remove it from `SqlNormalizedCache`:  `com.apollographql.apollo.internal.interceptor.ApolloCacheInterceptor#interceptAsync`

Why

Reply:
Hi @sav007 

I am picking this one up again.

Have been looking in the code on how to implement it higher up in the code, but cannot find how to do it.

Can we merge it like this? It fixes a critical bug for our application where we need to control which queries are NOT cached.